install_config/certificate_customization.adoc

```diff
@@ -19,82 +19,277 @@ This can be done during an
+
+However, by default, the registry uses routes to allow external systems and users to do pulls and pushes of images. You can use a _re-encrypt route_ with a custom certificate that will be presented to external users instead of using the internal, self-signed certificate. 
+
+To configure this, add the following lines *_/etc/ansible/hosts_* file in the *[OSEv3:vars]*
+section. pointing to the certificates you would like used for the registry route.
+
+----
+openshift_hosted_registry_routehost=registry.apps.c1-ocp.myorg.com <1>
+openshift_hosted_registry_routecertificates={"certfile": "/path/to/registry.apps.c1-ocp.myorg.com.crt", "keyfile": "/path/to/registry.apps.c1-ocp.myorg.com.key", "cafile": "/path/to/registry.apps.c1-ocp.myorg.com-ca.crt"} <2>
+----
+
+//https://github.com/openshift/openshift-docs/issues/5765
+<1> The host name of the registry
+<2> The locations of the *cacert*, *root*, and *cafile* files.
++
+* *certfile* contains the {product-title} custom router certificate.
+
+* *keyfile* contains the {product-title} custom key
+
+* *cafile* contains the root CA for this key and certificate. If an intermediate CA is in use it should contain both the intermediate and root CA.
+
+[[configuring-custom-certificates-retrofit]]
+=== Retrofit Custom Certificates into a Cluster
+
+To retrofit custom certificates into an existing {product-title} cluster: 
+
+. Edit the Ansible inventory file *_/etc/ansible/hosts_* to set the `openshift_master_overwrite_named_certificates=true` and specify the path to the certificate using `openshift_master_named_certificates`. 
++
+[source,yaml]
+----
+openshift_master_overwrite_named_certificates=true
+openshift_master_named_certificates=[{"certfile": "/path/on/host/to/crt-file", "keyfile": "/path/on/host/to/key-file", "names": ["public-master-host.com"], "cafile": "/path/on/host/to/ca-file"}] <1>
+----
++
+<1> Path to a xref:configuring-custom-certificates-master[master API certificate]. 
+
+. Run the following playbook:
++
+----
+ansible-playbook /usr/share/ansible/openshift-ansible/playbooks/byo/openshift-cluster/redeploy-master-certificates.yml
+----
+
+To retrofit custom router certificates into an existing cluster: 
+
+. Edit the Ansible inventory file *_/etc/ansible/hosts_* to set the `openshift_master_overwrite_named_certificates=true` and specify the path to the certificate using `openshift_hosted_router_certificate`
++
+[source,yaml]
+----
+openshift_master_overwrite_named_certificates=true
+openshift_hosted_router_certificate=[{"certfile": "/path/on/host/to/app-crt-file", "keyfile": "/path/on/host/to/app-key-file", "cafile": "/path/on/host/to/app-ca-file"}] <1>
+----
++
+<1> Path to a xref:configuring-custom-certificates-wildcard[wildcard API certificate].
+
+. Run the following playbook:
++
+----
+ansible-playbook /usr/share/ansible/openshift-ansible/playbooks/byo/openshift-cluster/redeploy-router-certificates.yml
```

Comment:
And for all the playbooks, if it's for ocp-3.9, the path of playbooks are all changed.